### PR TITLE
fix http sniffer logic in AP manager

### PR DESCRIPTION
**Pull request by matthyx**
restoring some formatting changes and deleted comments that were part of the previous PR
adding restoring of endpoints in case of error
adding reporting of dropped http event
renaming some fields for consistency

### CODE REVIEW

File: pkg/applicationprofilemanager/v1/applicationprofile_manager.go [hunk 1]
```diff
@@ -501,6 +499,13 @@ func (am *ApplicationProfileManager) saveProfile(ctx context.Context, watchedCon
 		if gotErr != nil {
 			// restore capabilities set
 			am.toSaveCapabilities.Get(watchedContainer.K8sContainerID).Append(capabilities...)
+			// restore endpoints map entries
```
Comment (matthyx) [inline pkg/applicationprofilemanager/v1/applicationprofile_manager.go:502]: you forgot this

File: pkg/applicationprofilemanager/v1/applicationprofile_manager.go [hunk 2]
```diff
@@ -527,19 +539,11 @@ func (am *ApplicationProfileManager) saveProfile(ctx context.Context, watchedCon
 			})
 			// record saved opens
 			toSaveOpens.Range(utils.SetInMap(am.savedOpens.Get(watchedContainer.K8sContainerID)))
-			// record saved endpoints
-			toSaveHttpEndpoints.Range(func(path string, endpoint *v1beta1.HTTPEndpoint) bool {
-				if !am.savedHttpEndpoints.Get(watchedContainer.K8sContainerID).Has(path) {
```
Comment (matthyx) [inline pkg/applicationprofilemanager/v1/applicationprofile_manager.go:532]: I don't think we need to check that

File: pkg/containerwatcher/v1/container_watcher.go [hunk 3]
```diff
@@ -320,15 +320,20 @@ func CreateIGContainerWatcher(cfg config.Config, applicationProfileManager appli
...
 	httpWorkerPool, err := ants.NewPoolWithFunc(httpWorkerPoolSize, func(i interface{}) {
 		event := i.(tracerhttptype.Event)
+		// ignore events with empty container name
 		if event.K8s.ContainerName == "" {
 			return
 		}
 
 		k8sContainerID := utils.CreateK8sContainerID(event.K8s.Namespace, event.K8s.PodName, event.K8s.ContainerName)
 
+		if isDroppedEvent(event.Type, event.Message) {
```
Comment (matthyx) [inline pkg/containerwatcher/v1/container_watcher.go:330]: added this reporting

File: pkg/applicationprofilemanager/v1/applicationprofile_manager.go [hunk 4]
```diff
@@ -295,17 +304,6 @@ func (am *ApplicationProfileManager) saveProfile(ctx context.Context, watchedCon
 		opens[path].Append(open.ToSlice()...)
 		return true
 	})
-
-	endpoints := make(map[string]*v1beta1.HTTPEndpoint)
-
-	toSaveHttpEndpoints := am.toSaveHttpEndpoints.Get(watchedContainer.K8sContainerID)
-	am.toSaveHttpEndpoints.Set(watchedContainer.K8sContainerID, new(maps.SafeMap[string, *v1beta1.HTTPEndpoint]))
-	toSaveHttpEndpoints.Range(func(path string, endpoint *v1beta1.HTTPEndpoint) bool {
-		if _, exist := endpoints[path]; !exist {
```
Comment (matthyx) [inline pkg/applicationprofilemanager/v1/applicationprofile_manager.go:304]: sorry, it was here that I was reporting the extra check

**Review by afek854 — APPROVED** [review]

**Pull request merged**